src/lib.rs

```diff
@@ -182,36 +179,49 @@ impl<S, C> AstarteDeviceSdk<S, C> {
                 mapping: path.to_string(),
             })?;
 
-        let (data, timestamp) = self.connection.deserialize_individual(mapping, payload)?;
+        let individual = self.connection.deserialize_individual(&mapping, payload)?;
 
-        if interface.is_property() {
-            let interface_name = interface.interface_name();
-            let path = path.as_str();
-            let interface_major = interface.version_major();
+        match individual {
```

Comment:
you can change this into `let...else {}`

Reply:
There are cases for both match arms, returning different values, here I like the match statment the most